File: src/main/java/fr/ribesg/blob/command/util/ShortenCommand.java
/*
 * Copyright (c) 2012-2014 Ribesg - www.ribesg.fr
 * This file is under GPLv3 -> http://www.gnu.org/licenses/gpl-3.0.txt
 * Please contact me at ribesg[at]yahoo.fr if you improve this file!
 */

package fr.ribesg.blob.command.util;
import fr.ribesg.alix.api.Channel;
import fr.ribesg.alix.api.Log;
import fr.ribesg.alix.api.Receiver;
import fr.ribesg.alix.api.Server;
import fr.ribesg.alix.api.Source;
import fr.ribesg.alix.api.bot.command.Command;
import fr.ribesg.alix.api.bot.command.CommandManager;
import fr.ribesg.alix.api.bot.util.WebUtil;
import fr.ribesg.alix.api.enums.Codes;

import java.io.IOException;

public class ShortenCommand extends Command {

   public ShortenCommand() {
      super("shorten", new String[] {
         "Shorten a url with the http://is.gd/ api",
         "Usage: ## <url>"
      }, "s");
   }

   @Override
   public boolean exec(final Server server, final Channel channel, final Source user, final String primaryArgument, final String[] args) {
      final Receiver receiver = channel == null ? user : channel;

      if (args.length != 1) {
         return false;
      }

      final String url = args[0];
      try {
         final String shortUrl = WebUtil.shortenUrl(url);
         receiver.sendMessage("Done: " + Codes.LIGHT_GREEN + shortUrl);
      } catch (final IOException e) {
         receiver.sendMessage(Codes.RED + "Failed to shorten URL");
         Log.error("Failed to shorten URL", e);
      }
      return true;
   }
}

File: src/main/kotlin/fr/ribesg/blob/command/util/ShortenCommand.kt
package fr.ribesg.blob.command.util

import fr.ribesg.blob.command.Command
import fr.ribesg.blob.extensions.sendRedNotice
import fr.ribesg.blob.util.Codes
import fr.ribesg.blob.util.Log
import fr.ribesg.blob.util.WebUtil
import org.kitteh.irc.client.library.Client
import org.kitteh.irc.client.library.element.Channel
import org.kitteh.irc.client.library.element.User
import java.io.IOException

/**
 * @author Ribesg
 */

public class ShortenCommand(prefix: Char) : Command(prefix) {

    init {
        this.usage = this.prefix + "s <url>"
    }

    override fun exec(client: Client, from: User, to: Channel?, primArg: String?, args: Array<String>) {
        if (args.size() != 1) {
            this.sendUsage(from)
        } else {
            val url = args[0];
            try {
                val shortUrl = WebUtil.shortenUrl(url);
                if (to == null) {
                    from.sendMessage("Result: " + shortUrl)
                } else {
                    to.sendMessage(from.getName() + ", " + shortUrl)
                }
            } catch (e: IOException) {
                from.sendRedNotice(Codes.RED + "Failed!")
                Log.error("Failed to shorten URL '" + url + "'", e);
            }
        }
    }
}
